packages/client/src/lib.js

```diff
@@ -70,18 +70,26 @@ class NFTStorage {
   /**
    * @param {API.Service} service
    * @param {Blob} blob
+   * @param {boolean} [isCar=false]
    * @returns {Promise<API.CIDString>}
    */
-  static async storeBlob({ endpoint, token }, blob) {
+  static async storeBlob({ endpoint, token }, blob, isCar = false) {
     const url = new URL('/upload', endpoint)
 
     if (blob.size === 0) {
       throw new Error('Content size is 0, make sure to provide some content')
     }
 
+    let headers = NFTStorage.auth(token)
+    if (isCar) {
+      headers = Object.assign(headers, {
+        'Content-Type': 'application/car',
+      })
+    }
```

Comment:
as `/upload` enpoint is using the `content-type` to determine how to process the file, it seemed safer to ensure `application/car` was used here. If the user creates a CAR somewhere else and then adds it via `<input type="file">` they need to be able to indicate it should be treated as a CAR, even if the blob's content type is not `application/car`. So we either require that callers re-encode their blobs to set the content-type, or re-create the blob here if it's type is not car, or we just force the header here. What do you think @Gozala?

Reply:
I've gone ahead and made it re-encode the blob to force the type to be `application/car`

Reply:
i've removed the code to explicitly set the request header, but we now have a weirdness where if a user provides a web-std Blob instead of a Blob (by what terrible accident they do that is a seperate issue, but is the case for ipfs-car bundled with vite at the moment) then the browser does not feel obliged to set the request header to match the type of a Blob-like thing.